### ISSUE TITLE
Chained DC2 adds static unless VU between each

### ISSUE THREAD

**Issue by DaveBenham** [sumpygump/quantal-audio#2]
With the new expander based daisy mixers, all works well if I add a VU for each DC2.

But if I chain two or more DC2 without any VU, then I get significant static or noise added.

I am on Win 10, running Rack 2.2.3

Also, the VU probably should work with the D-MX2 as well

**Comment by sumpygump** [comment]
Thanks for the report. I'll check it out. If possible can you send a patch where the issue is happening for you?

Good idea about the VU working with the D-MX2 too. I will add that.

**Comment by DaveBenham** [comment]
It is worse than I thought. The VU meter sometimes helps, sometimes not. And even the last DC2 in the chain has intermittent clicks. But the earlier DC2 are full of noise. I'm guessing you have a significant problem with your expander implementation.

The patch is about as simple as you can get. You should be able to reproduce using the image.

Green trace is the original signal from the VCO. Pink is after going through the mixer.

![DaisyNoise](https://user-images.githubusercontent.com/17756633/222933589-e650c78e-01ac-4e5b-beba-6fe75efeafd6.PNG)

**Comment by sumpygump** [comment]
Thanks for the details. Yeah, that looks bad. I cannot recreate it on Linux; I'll see what I get when running on Windows.

**Comment by sumpygump** [comment]
I racked my brain trying to figure out the cause; finally resolved it!

I verified the problem only existed in Windows (worked fine in Linux and MacOS), which led me to believe the issue was based on some way that floating point calculations were being done in Windows, but none of my attempts were producing good results. Then I finally figured out it was not an issue with Windows, but an issue with threading. If Rack engine is set to use more than one thread, the issue was present. Now I fixed it so the expander signal flow is thread-safe.

I also added support for the VU meter being attached to the master daisy channel  too.
